<div style="border:solid Chocolate 2px; padding: 40px">

<b>Привет!</b>

Меня зовут Дмитрий Михаленко, я буду ревьюером твоего проекта. Если ты не против, то предлагаю построить наше общение на "ты" ;) Если удобнее на "вы", то нет проблем, только скажи об этом. 

Ниже в работе ты найдешь мои комментарии - **пожалуйста, не перемещай, не изменяй и не удаляй их**.
    
Для наглядности я буду использовать цветовую разметку:
<font color='green'>зеленого</font>, <font color='gold'>желтого</font> и <font color='red'>красного</font> цветов. Сейчас объясню, что они значат:

<br/>

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Решение на этом шаге является полностью правильным.
</div>

<br/>

<div class="alert alert-warning">
    <h2> Комментарий ревьюера <a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b> Решение на этом шаге может стать еще лучше, если внести небольшие коррективы.
Но настаивать я не буду.
</div>


<br/>
<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера <a class="tocSkip"></h2>

    
<b>На доработку🤔:</b>
Для правильного решения задачи на этом шаге необходимо внести правки. Напоминаю, что проект не может быть принят, без их отработки.
</div>

Для продуктивной работы прошу: **если ты что-то меняешь в проекте или отвечаешь на мои комментарии — пиши об этом**
Мне будет легче отследить изменения, если ты тоже выделишь свои комментарии:
    
<div class="alert alert-info"> <b>Комментарий студента:</b> Например, вот так.</div>

Всё это поможет выполнить повторную проверку твоего проекта оперативнее. 
        
Увидев у тебя ошибку, в первый раз я лишь укажу на ее наличие и дам тебе возможность самому найти и исправить ее. На реальной работе твой начальник будет поступать так же, а я пытаюсь подготовить тебя именно к работе аналитиком. Но если ты пока не справишься с такой задачей - при следующей проверке я дам более точную подсказку.

# Анализ игр разных жанров

- Автор: Воронкова Дарья Игоревна
- Дата: 12.02.2025

### Цели и задачи проекта

<font color='#777778'>В этой тетрадке используем датасет new_games.csv, который содержит информацию о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр. При знакомстве с данными проверим корректность типов данныхБ а так же пропуски и явные и неявные дубликаты. После отчистки и удаления дубликатов по проекту будут стоять следующие задачами:
- Категоризация игр по оценкам пользователей и экспертов;
- Топ-7 платформ по количеству игр, выпущенных за период с 2000 по 2013 год включительно.</font>

### Описание данных

 В проекте будут использованы данные датасета /datasets/new_games.csv с таким описанием::
- Name — название игры.
- Platform — название платформы.
- Year of Release — год выпуска игры.
- Genre — жанр игры.
- NA sales — продажи в Северной Америке (в миллионах проданных копий).
- EU sales — продажи в Европе (в миллионах проданных копий).
- JP sales — продажи в Японии (в миллионах проданных копий).
- Other sales — продажи в других странах (в миллионах проданных копий).
- Critic Score — оценка критиков (от 0 до 100).
- User Score — оценка пользователей (от 0 до 10).
- Rating — рейтинг организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.

### Содержимое проекта
- [Загрузка данных и знакомство с ними](#1-bullet)
- [Проверка ошибок в данных и их предобработка](#2-bullet)
- [Фильтрация данных](#3-bullet)
- [Категоризация данных](#4-bullet)
- [Итоговые выводы](#5-bullet)
---

<div class="alert alert-success">
<h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b>  Отличное введение в проект! Есть цель работы, описание исходных  данных  и содержание. 
    
Такой детальный подход важен, чтобы твой будующий работодатель знал все ключевые моменты проекта и мог объективно оценить принимаемые тобой решения.
    


<a class="anchor" id="1-bullet"></a>
## 1. Загрузка данных и знакомство с ними

- Загрузим необходимые библиотеки для анализа данных и данные из датасета `new_games.csv`. Затем выведем основную информацию о данных с помощью метода `info()` и первые строки датафрейма.


In [30]:
# Импортируем библиотеку pandas
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

In [31]:
# Выгружаем данные из датасета new_games.csv в датафрейм games
games = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')

- Познакомьтесь с данными: выведите первые строки и результат метода `info()`.


In [32]:
# Выводим информацию о датафрейме
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


In [33]:
# Выводим первые строки датафрейма на экран
games.head()

Name Platform  Year of Release         Genre  NA sales  \
0                Wii Sports      Wii           2006.0        Sports     41.36   
1         Super Mario Bros.      NES           1985.0      Platform     29.08   
2            Mario Kart Wii      Wii           2008.0        Racing     15.68   
3         Wii Sports Resort      Wii           2009.0        Sports     15.61   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing     11.27   

  EU sales JP sales  Other sales  Critic Score User Score Rating  
0    28.96     3.77         8.45          76.0          8      E  
1     3.58     6.81         0.77           NaN        NaN    NaN  
2    12.76     3.79         3.29          82.0        8.3      E  
3    10.93     3.28         2.95          80.0          8      E  
4     8.89    10.22         1.00           NaN        NaN    NaN

Датасет `new_games.csv` содержит 11 столбцов и 16956 строк, в которых представлена информация о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр.

Изучим типы данных и их корректность:
- **Числовые значения с плавающей запятой (float64).** Четыре столбца имеют тип данных `float64`. Это `Year of Release`, `NA sales`, `Other sales `, `Critic Score `. 
    - `Year of Release` хранит данные о годе выпуска игры. Тут лучше использовать тип `int64`, однако можно уменьшить разрядность до `int32` или `int16`, если значения достаточно малы и это не повлияет на точность.
- **Строковые данные (object).** Семь столбцов имеют тип данных `object`:
    - `Name`, `Platform`,`Genre` и `Rating` содержат строковую информацию (название игры, платформа и жанр, а так же рейтинг организации ESRB), что логично для текстовых данных. Здесь тип данных `object` подходит.
    - `EU sales`, `JP sales` и `User Score` хранят информацию о продажах в Европе и Японии, а так же оценку пользователей. Для таких данных рекомендуется использовать тип `float64`, который включает дробные значения и облегчит выполнение математических вычислений.
    

После анализа типов данных видно, что большинство столбцов представлены в нужном типе данных. Однако для корректной работы с данными нужно изменить формат данных.

<div class="alert alert-success">
<h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b>  Хороший выбор методов для первичной оценки данных! Молодец, что обращаешь внимание на возможности оптимизации типов данных для этог проекта

---
<a class="anchor" id="2-bullet"></a>
## 2.  Проверка ошибок в данных и их предобработка


### 2.1. Названия, или метки, столбцов датафрейма

- Выведите на экран названия всех столбцов датафрейма и проверьте их стиль написания.
- Приведите все столбцы к стилю snake case. Названия должны быть в нижнем регистре, а вместо пробелов — подчёркивания.

In [34]:
# Выводим название всех столбцов в датафрейме:
games.columns

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='object')

In [39]:
# Приведем название всех столбцов в датафрейме  к стилю snake case:
games.columns=games.columns.str.lower().str.replace(' ', '_')

Все столбцы приведены к стилю snake case и перезаписаны в датафрейм new_games.csv. Названия в нижнем регистре, а вместо пробелов — подчёркивания.

<div class="alert alert-warning">
    <h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b>  Полезная привычка следовать "змеиному стилю". Можно для этого также использовать цепочный метод `games.columns.str.lower().str.replace(' ', '_')`

<div class="alert alert-info"> <b>Комментарий:</b> Внесла изменения в код.</div>

<div class="alert alert-success">
<h2> Комментарий ревьюера v2 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b>Рад, что эта идея показалась тебе полезной)

### 2.2. Типы данных

- Если встречаются некорректные типы данных, предположите их причины.
- При необходимости проведите преобразование типов данных. Помните, что столбцы с числовыми данными и пропусками нельзя преобразовать к типу `int64`. Сначала вам понадобится обработать пропуски, а затем преобразовать типы данных.

In [41]:
# Сохраняем общее количество строк датафрейма для дальнейшего анализа
games_lines = games.shape[0]

In [42]:
# Приобразуем тип данных у колонки 'year_of_release'к типу 'int64' и уменьшим разрядность и удалим пропуски:
games = games.dropna(subset=['year_of_release'])
games['year_of_release'] = games['year_of_release'].astype ('int64')
games['year_of_release'] = pd.to_numeric(games['year_of_release'], downcast='integer')

У колонки `year_of_release` изначальный тип данных был `float64`. Причины неккорекного типа данных возможна была в том, что тип `float64` занимает меньше  памяти. Либо при сборе данных была допущена ошибка записи данных. Для того, чтобы преобразовать колонку `year_of_release` вначале пришлось удалить пропушенные значения, т.к. всего 1,62% данных были пропущены. Так же для экономии памяти тип данных `int64` был преобразован в тип `int16` 

- В числовых столбцах могут встретиться строковые значения, например `unknown` или другие. Приводите такие столбцы к числовому типу данных, заменив строковые значения на пропуски.

In [43]:
# Приобразуем тип данных у колонок 'eu_sales', 'jp_sales', 'user_score'к типу 'float64':
games[['eu_sales', 'jp_sales', 'user_score']] = games[['eu_sales', 'jp_sales', 'user_score']].replace ('unknown', np.nan)
games[['eu_sales', 'jp_sales', 'user_score']] = games[['eu_sales', 'jp_sales', 'user_score']].replace ('tbd', np.nan)
games[['eu_sales', 'jp_sales', 'user_score']] = games[['eu_sales', 'jp_sales', 'user_score']].astype ('float64')

In [44]:
games.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16681 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16679 non-null  object 
 1   platform         16681 non-null  object 
 2   year_of_release  16681 non-null  int16  
 3   genre            16679 non-null  object 
 4   na_sales         16681 non-null  float64
 5   eu_sales         16675 non-null  float64
 6   jp_sales         16677 non-null  float64
 7   other_sales      16681 non-null  float64
 8   critic_score     8085 non-null   float64
 9   user_score       7558 non-null   float64
 10  rating           9901 non-null   object 
dtypes: float64(6), int16(1), object(4)
memory usage: 1.4+ MB


У колонок `eu_sales`, `jp_sales`, `user_score` преобразовали тип данных к типу `float64`. Некорректные типы данных были связаны с тем, что в этих столбцах присутствовали строкоые значения. Данные значения были заменены на 'Nan' для корректного преобразования к нужному типу данных

<div class="alert alert-success">
<h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Последовательная и качественная работа с преобразованием типов данных, молодец!

### 2.3. Наличие пропусков в данных

- Посчитайте количество пропусков в каждом столбце в абсолютных и относительных значениях.


In [51]:
# Выводим количество пропущенных строк в датафрейме
games.isna().sum().sort_values(ascending=False)

user_score         9123
critic_score       8596
rating             6780
eu_sales              6
jp_sales              4
name                  2
genre                 2
platform              0
year_of_release       0
na_sales              0
other_sales           0
dtype: int64

In [52]:
# Подсчитываем процент строк с пропусками
games.isna().mean().sort_values(ascending=False)

user_score         0.546910
critic_score       0.515317
rating             0.406450
eu_sales           0.000360
jp_sales           0.000240
name               0.000120
genre              0.000120
platform           0.000000
year_of_release    0.000000
na_sales           0.000000
other_sales        0.000000
dtype: float64

<div class="alert alert-warning">
    <h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b> Отличная идея изучать, не только количество, но и доли пропусков! Предложу как  сделать это немного лаконичнее: использовать метод `mean()` после вызова `isna()` 
        
Также, полезно применять для таких таблиц сортировку методом `sort_values()`

<div class="alert alert-info"> <b>Комментарий:</b> Поменяла формулу.</div>

В колонах `name`, `genre`, `eu_sales` и `jp_sales`  пропушенных значений не более 6 шт. Эти пропуски могли возникнуть из-за человеческого фактора, т.к. процент этих пропусков очень мал (0,01%) данные в этих столбцах можно просто удалить.  
В колонах `critic_score `, `user_score` и `rating` отсутсвует более 50% данных. Данный тип пропусков относится к группе MNAR.
Эти пропуски не случайны, но связаны с факторами, которых нет в наборе данных. К ним могут относиться и сами пропущенные значения.
Заполним пропуски `critic_score `, `user_score` с помощью значения-индикатора. В качестве значения выберем -1, т.к. это значение не встречается внаборе данных. `rating` оставим без изменений, т.к. в анализе данных он не принимает участие.

- Обработайте пропущенные значения. Для каждого случая вы можете выбрать оптимальный, на ваш взгляд, вариант: заменить на определённое значение, оставить как есть или удалить.
- Если вы решите заменить пропуски на значение-индикатор, то убедитесь, что предложенное значение не может быть использовано в данных.
- Если вы нашли пропуски в данных с количеством проданных копий игры в том или ином регионе, их можно заменить на среднее значение в зависимости от названия платформы и года выхода игры.

In [55]:
# Удалим пропуски у колонок 'name','genre', 'eu_sales' и 'jp_sales' :
games = games.dropna(subset=['name', 'genre', 'eu_sales', 'jp_sales'])

<div class="alert alert-success">
<h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b>Согласен, небольшой объем данных вполне можно удалить

In [59]:
#В колонах 'critic_score'и 'user_score' заменим пропуски на значение индикатор:
games [['critic_score','user_score']]=games [['critic_score','user_score']].fillna(-1)

<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера v1<a class="tocSkip"></h2>

<b>На доработку🤔:</b> Дарья, функции реализовала классно! Но не уверен, что такая замена не внесет искажений. Игровая сфера, во многом зависит от бюджета проекта и успешности его рекламы. Сложно учитывать из чего будут складываться оценки игры, это и жанр с платформой, как ты справедливо заметила, так и год релиза и охват аудитории.
    
Предлагаю в этом вопросе остановиться на "заглушках"

<div class="alert alert-info"> <b>Комментарий:</b> Произвела замену на значение индикатор -1.</div>

<div class="alert alert-success">
<h2> Комментарий ревьюера v2 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b>Практичное решение, обращу внимание, что при работе с "заглушками" важно не забывать убирать их перед расчетом статистичнских показателей, чтобы не внести искажения

In [61]:
games.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16669 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16669 non-null  object 
 1   platform         16669 non-null  object 
 2   year_of_release  16669 non-null  int16  
 3   genre            16669 non-null  object 
 4   na_sales         16669 non-null  float64
 5   eu_sales         16669 non-null  float64
 6   jp_sales         16669 non-null  float64
 7   other_sales      16669 non-null  float64
 8   critic_score     16669 non-null  float64
 9   user_score       16669 non-null  float64
 10  rating           9892 non-null   object 
dtypes: float64(6), int16(1), object(4)
memory usage: 1.4+ MB


Для дальнейшего анализа данные частично подготовлены. Осталось отчистить данные от дубликатов и приступить к исследованию 

### 2.4. Явные и неявные дубликаты в данных

- Изучите уникальные значения в категориальных данных, например с названиями жанра игры, платформы, рейтинга и года выпуска. Проверьте, встречаются ли среди данных неявные дубликаты, связанные с опечатками или разным способом написания.
- При необходимости проведите нормализацию данных с текстовыми значениями. Названия или жанры игр можно привести к нижнему регистру, а названия рейтинга — к верхнему.

In [62]:
# Найдем уникальные значения в таких колонках, как 'name','genre','platform' и 'year_of_release':
unique_name = games['name'].nunique() 
print (unique_name)

11422


In [63]:
unique_genre = games['genre'].unique() 
print (unique_genre)

['Sports' 'Platform' 'Racing' 'Role-Playing' 'Puzzle' 'Misc' 'Shooter'
 'Simulation' 'Action' 'Fighting' 'Adventure' 'Strategy' 'MISC'
 'ROLE-PLAYING' 'RACING' 'ACTION' 'SHOOTER' 'FIGHTING' 'SPORTS' 'PLATFORM'
 'ADVENTURE' 'SIMULATION' 'PUZZLE' 'STRATEGY']


In [64]:
unique_platform = games['platform'].unique()
print (unique_platform)

['Wii' 'NES' 'GB' 'DS' 'X360' 'PS3' 'PS2' 'SNES' 'GBA' 'PS4' '3DS' 'N64'
 'PS' 'XB' 'PC' '2600' 'PSP' 'XOne' 'WiiU' 'GC' 'GEN' 'DC' 'PSV' 'SAT'
 'SCD' 'WS' 'NG' 'TG16' '3DO' 'GG' 'PCFX']


In [65]:
unique_year = games['year_of_release'].unique()
print (unique_year)

[2006 1985 2008 2009 1996 1989 1984 2005 1999 2007 2010 2013 2004 1990
 1988 2002 2001 2011 1998 2015 2012 2014 1992 1997 1993 1994 1982 2016
 2003 1986 2000 1995 1991 1981 1987 1980 1983]


In [66]:
# Проведем нормализацию данных и приведем текстовые значения к нижнему регистру:
games['genre'] = games['genre'].str.lower() 

In [67]:
games['name'] = games['name'].str.lower() 

In [68]:
games['rating'] = games['rating'].str.upper() 

- После того как нормализуете данные и устраните неявные дубликаты, проверьте наличие явных дубликатов в данных.

In [69]:
# Сортируем датафрейм по всем столбцам:
games_sorted = games.sort_values(by=games.columns.tolist())

In [70]:
# Находим дубликаты
duplicates = games_sorted[games_sorted.duplicated(keep='first')]
print (duplicates)

                                            name platform  year_of_release  \
15192                             beyblade burst      3DS             2016   
15302                          11eyes: crossover     X360             2009   
4861            18 wheeler: american pro trucker      PS2             2001   
13099                                 4 elements       PC             2009   
5236   999: nine hours, nine persons, nine doors       DS             2009   
...                                          ...      ...              ...   
16104             yoake yori ruriiro na portable      PSP             2010   
10662            yu-gi-oh! 5d's wheelie breakers      Wii             2009   
2909            yu-gi-oh! the falsebound kingdom       GC             2002   
6696                               zoo resort 3d      3DS             2011   
8157                          zumba fitness rush     X360             2012   

              genre  na_sales  eu_sales  jp_sales  other_sales 

In [71]:
# Найдем количество дубликатов в Датафрейме:
duplicated_games = games.duplicated().sum()
print (duplicated_games)

235


In [72]:
# Удаляем дубликаты:
game_no_duplicates = games_sorted.drop_duplicates()

In [73]:
# Сохраняем количество строк после удаления дубликатов
final_count = game_no_duplicates.shape[0]

- Напишите промежуточный вывод: укажите количество найденных дубликатов и действия по их обработке.

После использования строковых методов по выявлению неявных дубликатов и применению их в столбцам датасета `new_games.csv` было выявлено и удалено 235 явных дубликатов.

- В процессе подготовки данных вы могли что-либо удалять, например строки с пропусками или ошибками, дубликаты и прочее. В этом случае посчитайте количество удалённых строк в абсолютном и относительном значениях.

In [74]:
# Найдем количество удаленных строк
count_games= games_lines - final_count
print (count_games)

522


In [75]:
# Найдем долю удаленных строк
percentage_values = count_games/games_lines * 100
print (round (percentage_values,2))

3.08


<div class="alert alert-success">
<h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Процент удаления не большой и исследование не потеряет своей объективности!

- После проведения предобработки данных напишите общий промежуточный вывод.

Были загружены данные `new_games.csv`.Они содержат 11 столбцов и 16956 строк, в которых представлена информация о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр. При первичном знакомстве с данными и их предобработкой получили такие результаты:
- В семи столбцах были обнаружены пропущенные значения. Максимальное значение пропущенных данных в столбце `user_score ` — 54,7%;
- Для оптимизации работы с данными в датафрейме были произведены изменения типов данных;
- Названия столбцов были приведены к стилю snake case;
- Пропуски, где процент был менее 5, были удалены, т.к. отклонением данных менее 5% можно принебреч;
- Пропуски, где был высокий процент, были заменены на значение индикатор;
- Столбец `rating` остался без изменения данных, т.к. в дальнейшем анализе он не потребуется;
- Было выявлено и удалено 235 явных и не явных дубликата;
- После всех преобразований количество данных уменьшилось на 522 строки или на 3,1%.

<div class="alert alert-success">
<h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b>Дарья, этап в целом выполнен хорошо, ты внимательна к деталям и развернуто комментируешь результаты, хвалю!

---
<a class="anchor" id="3-bullet"></a>
## 3. Фильтрация данных

Коллеги хотят изучить историю продаж игр в начале XXI века, и их интересует период с 2000 по 2013 год включительно. Отберите данные по этому показателю. Сохраните новый срез данных в отдельном датафрейме, например `df_actual`.

In [76]:
# Отфильтруем данные за период 2000 по 2013 год включительно и сохрвним в новый датафрейм 'df_actual':
df_actual = games.query('2000 <= year_of_release <= 2013')

В новом датафреме `df_actual` хранятся данные о продаж игр за период с 2000 по 2013 год включительно, которые понадобятся для дальнейшего решения дальнейших задач проекта

<div class="alert alert-warning">
    <h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b>  Фильтрация проведена корректно. Покажу альтернативный вариант 
        
        df_actual = df.query('2000 <= year_of_release <= 2013')


<div class="alert alert-info"> <b>Комментарий:</b> Внесла изменения в формулу.</div>

---
<a class="anchor" id="4-bullet"></a>
## 4. Категоризация данных
    
Проведите категоризацию данных:
- Разделите все игры по оценкам пользователей и выделите такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

In [77]:
# Разбиваем все игры на категории по оценкам пользователей:
df_actual['user_group'] = pd.cut(df_actual['user_score'], bins=[0, 3, 8, 10], labels=["низкая оценка", "средняя оценка", "высокая оценка"], right=False)

- Разделите все игры по оценкам критиков и выделите такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [78]:
# Разбиваем все игры на категории по оценкам критиков:
df_actual['critic_group'] = pd.cut(df_actual['critic_score'], bins=[0, 30, 80, 100], labels=["низкая оценка", "средняя оценка", "высокая оценка"], right=False)

- После категоризации данных проверьте результат: сгруппируйте данные по выделенным категориям и посчитайте количество игр в каждой категории.

In [79]:
# Вычеслим количество оценок по категориям (оценки пользователей):
user_groups = df_actual.groupby('user_group')['name'].count()
print (user_groups)

user_group
низкая оценка      117
средняя оценка    4145
высокая оценка    2303
Name: name, dtype: int64


In [80]:
# Вычеслим количество оценок по категориям (оценки критиков):
critic_groups = df_actual.groupby('critic_group')['name'].count()
print (critic_groups)

critic_group
низкая оценка       55
средняя оценка    5499
высокая оценка    1706
Name: name, dtype: int64


Большое количество оценок пользователей и критиков находится в группе средняя оценка. Можно сделать вывод, что игры, выпущенные в переод с 2000 по 2013 год включительно были среднего качества и не дотягивали до наивысших оценнок. В это период индутрия игр только набирала обороты и технологии комппьютерных игр также находилась на этапе развития

- Выделите топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [81]:
# Считаем количество игр для каждой платформы за период с 2000 по 2013 год включительно
platform_counts= df_actual['platform'].value_counts()

In [82]:
# Выделяем топ-7 платформ:
top_7_platforms = platform_counts.head(7)
print (top_7_platforms)

PS2     2153
DS      2143
Wii     1294
PSP     1198
X360    1135
PS3     1107
GBA      825
Name: platform, dtype: int64


В анализируемом периоде платформы PS2 и DS занимают лидирующие позиции в рейтинге. 

<div class="alert alert-success">
<h2> Комментарий ревьюера v1 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> "Топ-7" определен корректно, ты отлично усвоила инструменты фильтрации и группировки, хвалю!

---
<a class="anchor" id="5-bullet"></a>
## 5. Итоговый вывод

В конце напишите основной вывод и отразите, какую работу проделали. Не забудьте указать описание среза данных и новых полей, которые добавили в исходный датасет.

В проекте `Аназиз игр разых жанров` мы использовали датасет `new_games.csv`, который содержит информацию о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр.Перед нами стояли следующие цели:
- Выявление ошибок в данных и их предобработка (удаление явных и не явных дубликатов, заполнение пропусков);
- Фильтрация данных за определенный период;
- Категоризация игр по оценкам пользователей и экспертов;
- Топ-7 платформ по количеству игр.
В ходе выполненного проекта были получены следующие результаты:
- В семи столбцах были обнаружены пропущенные значения; 
- Для оптимизации работы с данными в датафрейме были произведены изменения типов данных;
- Названия столбцов были приведены к стилю snake case;
- Пропуски, где процент был менее 5, были удалены, т.к. отклонением данных менее 5% можно принебреч;
- Пропуски, где был высокий процент, были заменены значением индикатором;
- Было выявлено и удалено 235 явных и не явных дубликата;
- После всех преобразований количество данных уменьшилось на 522 строки или на 3,1%.

На этапе `Загрузка данных и знакомство с ними` мы познакомились в общей картиной данных и выявили, что в датасете `new_games.csv` содержит 11 столбцов и 16956 строк. Изучили типы данных и их корректность и выявили, что большинство столбцов представлены в нужном типе данных. Однако для корректной работы мы изменили типы данных у таких столбцов, как : `year_of_release`, `eu_sales`, `jp_sales` и `user_score`.

Так же в процессе знакомства с данными были выявлены пропущенные значения часть которых была удалена (где количество пропусков было менее 5%).А часть была заменена на значение-индикатор -1 ( по колонкам `critic_score` и `user_score`).  
При анализе явных и не явных дубликатов было выявлено 235 дубликатов. После всех манипуляций с данными их количество уменьшилось на 522 шт. или на 3,1%.

В основном разделе анализа проекта была сделана фильтрация по годам и категоризация данных по оценкам пользователей и критиков. 
Так же выделили топ-7 платформ по количеству игр, выпущенных за период с 2000 по 2013 год включительно.

По итогом выполненного проекта было изучено развитие игровой индустрии. К сожалению данные, которые нашли коллеги в открытых источниках были неполными, но дастоточными для того, что бы сделать выводы для написания статьи о развитии индустрии игр в начале XXI века. Платформы PS2 и DS являются наиболее популярными в исследуемом периоде. 

<div class="alert alert-warning">
    <h2> Комментарий ревьюера v1<a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b> Дарья, итоговый вывод написан в хорошем стиле, наиболее важные результаты подробно и понятно резюмированы, основные шаги преобразований есть.
        
Итоговые выводы - важная часть работы и в твоих будующих реальных проектах советую уделять ей особое внимание. Бывают ситуации, когда у руководства нет времени вникать в весь проект и оценка твоих действий будет проводиться именно по ним.            
        
Пожалуйста, перепроверь их актуальность после устранения неточности выше. 

<div class="alert alert-info"> <b>Комментарий:</b> Актуальность выводов перепроверена и изменения внесены. Спасибо за высокую оценку моей работы. Делала ее два дня и потратила очень  много сил.</div>

<div style="border:solid Chocolate 2px; padding: 40px">
    
**Общий вывод по проекту ревьювера**:
     Спасибо за  проект! Ты приложила много усилий, чтобы довести его до конца, проделана огромная работа, и это видно невооруженным глазом, ты  молодец! Но у меня осталось несколько  дополнений с которыми проект может стать еще лучше.
    
**Отмечу положительные моменты**:
    
    1. Все разложено по полочкам, всегда понятен ход твоих мыслей, приятно смотреть.
    
    2. Подробные и логичные выводы на всех этапах работы.
    
    3. Внимание к деталям и учет особенностей бизнеса.      
    
**На что стоит обратить внимание**:
       
    1. Пересмотр принципа устранения пропусков данных
    
    2. Актуализация вывода
    
**Удачи и жду твой проект на повторное ревью!**
    
    

<div class="alert alert-success">
<h2> Комментарий ревьюера v2 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b>Дарья, спасибо за качественную отработку моих комментариев и хорошую обратную связь. Работа выполнена на очень высоком уровне! Код оптимизирован, выводы точны, аналитическая записка аккуратно оформлена.
    
Желаю тебе интересного и успешного продолжения учебы, увидимся в практикуме!